library/pk.c

```diff
@@ -150,11 +150,12 @@ int mbedtls_pk_setup( mbedtls_pk_context *ctx, const mbedtls_pk_info_t *info )
 /*
  * Initialise a PSA-wrapping context
  */
-int mbedtls_pk_setup_opaque( mbedtls_pk_context *ctx, const psa_key_handle_t key )
+int mbedtls_pk_setup_opaque( mbedtls_pk_context *ctx,
+                             const mbedtls_svc_key_id_t key )
 {
     const mbedtls_pk_info_t * const info = &mbedtls_pk_opaque_info;
     psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
-    psa_key_handle_t *pk_ctx;
+    mbedtls_svc_key_id_t *pk_ctx;
```

Comment:
Are you sure for (cipher¦pk)*.[hc] ? As PSA code is for example calling mbedtls_cipher_free() this is not really PSA client code, right?

Reply:
The cipher and pk modules currently contain both client-side code and server-side code because parts of them are above the PSA crypto API layer and parts are underneath. As part of the psa-isaztion of the library, all remaining calls to cipher and pk from `psa_crypto.c` should disappear eventually, and then cipher and pk will be client-side only. An opaque cipher or pk object is purely client-side.

Reply:
If MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER is set then the first argument of psa_get_key_attributes() (called by mbedtls_pk_setup_opaque()) is of type mbedtls_svc_key_id_t (!= psa_key_id_t) thus the 2nd parameter of mbedtls_pk_setup_opaque() as to be of type mbedtls_svc_key_id_t and pk_ctx as well. Thus if MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER  is set, I cannot move here to psa_key_id_t.

But, maybe it is just the case that mbedtls_pk_setup_opaque() should not be defined if MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER  is defined, thus incompatibility between  MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER  and MBEDTLS_USE_PSA_CRYPTO ?

Reply:
The cipher/pk opaque interface is intended as a transitional interface to facilitate the migration of existing code (such as the X.509 and TLS parts of Mbed TLS) to PSA. In a world where crypto is in its own service, it wouldn't need to exist on the server side. So indeed I don't have a use case for `MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER` and `MBEDTLS_USE_PSA_CRYPTO` together.

Reply:
Thus I guess that the way forward is for MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER not to be set in the full configuration as incompatible with MBEDTLS_USE_PSA_CRYPTO and add in all.sh a test component dedicated to MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER: full - MBEDTLS_USE_PSA_CRYPTO  + MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER where only the unit tests are run. Does it sound good?

Reply:
Actually [`MBEDTLS_USE_PSA_CRYPTO` should not be enabled in `full`](https://github.com/ARMmbed/mbedtls/issues/3603), so `MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER` could be. But in the short term, I guess it's better to keep `MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER` out of `full`, with a note that it should be added back once `MBEDTLS_USE_PSA_CRYPTO` is removed.

Reply:
I've removed MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER from the full configuration and added a component in all.sh to test the configuration (full - MBEDTLS_USE_PSA_CRYPTO + MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER).

Reply:
Addressed in commit:
lib: Remove usage of mbedtls_svc_key_id_t
State PSA_CRYPTO_KEY_ID_ENCODES_OWNER and USE_PSA_CRYPTO incompatibility